libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/util/config/ConfigManager.java

```diff
@@ -0,0 +1,463 @@
+            return this;
+        }
+
+        // Allow all config-providers to inspect the source-description and provide a config-source.
+        ConfigProvider.ConfigSource configSource = null;
+        for (ConfigProvider provider : configProviders) {
+            try {
+                configSource = provider.tryToBuildConfigSource(trimmedSourceDescription);
+                if (configSource != null) {
+                    break;
+                }
+            } catch (Exception exception) {
+                LOG.warn("Ignoring exception when config-provider {} tried to build config-source for source-description {}: ", provider, Utils.jsonSerializeForLog(trimmedSourceDescription), exception);
+            }
+        }
+        if (configSource == null) {
+            LOG.warn("No config-provider could provide a config-source for source-description {}. Ignoring.", Utils.jsonSerializeForLog(trimmedSourceDescription));
+            return this;
+        }
+
+        addConfigSource(configSource);
+        return this;
+    }
+
+
+    /**
+     * Add a config-source object - without "consulting" the {@link ConfigProvider}s.
+     * Sources that are added earlier have higher priority
+     *  (values from early-added sources prevail over values from later-added sources).
+     */
+    public synchronized ConfigManager addConfigSource(ConfigProvider.ConfigSource configSource) {
+
+        if (configSources.contains(configSource)) {
+            LOG.warn("Config-source {} added twice. Ignoring.", configSource);
+            return this;
+        }
+
+        // Add the config-source.
+        LOG.info("Added config-source: {}", configSource);
+        Set<ConfigProvider.ConfigSource> higherPrioritySources = new HashSet<>(configSources);
+        configSources.add(configSource);
+        List<ChangeLog> changeLogs = new LinkedList<>();
+        reloadConfigSource(configSource, higherPrioritySources, -1, changeLogs);
+
+        // Update System-properties, and call change-callbacks.
+        digestChangeLogs(changeLogs);
+        return this;
+    }
+
+    /** Undo a call to {@link #addConfigSource(String)} */
+    public synchronized boolean removeConfigSource(String sourceDescription) {
+        String trimmedSourceDescription = sourceDescription.trim();
+        boolean anySourceRemoved = configSources.removeIf(configSource -> configSource.sourceDescription.equals(trimmedSourceDescription));
+        if (anySourceRemoved) {
+            reloadAllConfigs();
+        }
+        return anySourceRemoved;
+    }
+
+    /** Undo a call to {@link #addConfigSource(ConfigProvider.ConfigSource)} */
+    public synchronized boolean removeConfigSource(ConfigProvider.ConfigSource configSource) {
+        boolean anySourceRemoved = configSources.remove(configSource);
+        if (anySourceRemoved) {
+            reloadAllConfigs();
+        }
+        return anySourceRemoved;
+    }
+
+    /** Get all config-sources */
```

Comment:
Please remove unnecessary comments. The function is called getConfigSources which is clear.

Reply:
I am almost with you, but the OCD part in me won't let me   :-(
Also - this is a package that may be used by many, so it needs to be well documented = 100%.
Also, arguably, if I don't comment SOME stuff, then I will become lazy and not comment MORE stuff.
I think that our take on comments are vastly different...